x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/AbstractPhysicalOperationProviders.java

```diff
@@ -172,6 +179,13 @@ else if (aggregatorMode == AggregatorMode.INITIAL || aggregatorMode == Aggregato
                     context
                 );
             } else {
+                if (groupSpecs.stream().anyMatch(g -> g.groupingFunction != null)) {
+                    // TODO: DONT DO THIS, bucket should work: assert groupSpecs.size() == 1 : "Only one grouping function is supported";
+                    // Maybe ignore Bucket directly with instanceof
+                    // Maybe add some method to GroupingFunction to get the hash?
+                    // Maybe a method returning a Supplier of HashBlock?
+                }
```

Comment:
Just adding a reminder here because that's an important `NOCOMMIT`.

Reply:
Removed! This was an initial idea I discarded, and forgot to delete it